src/global.d.ts

```diff
@@ -0,0 +1,3 @@
+interface Window {
```

Comment:
Prefer not to define it in `global.d.ts` because `ApplePaySession` isn't defined for all browsers. You can add a question mark there but I think a better way is to just define an interface like `ApplePayWindow` and a type guard `isApplePayWindow`. You can then use the type guard to check if `window` is a regular `Window` or `ApplePayWindow` in your `filterApplePay` method.